### PR TITLE
Internal: Editor V2 - User preferences [ED-9733]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: packages/packages/user-preferences/src/hooks/__tests__/use-action-props.test.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { openRoute } from '@elementor/v1-adapters';
+
+jest.mock( '@elementor/v1-adapters', () => ( {
+	openRoute: jest.fn(),
+} ) );
+
+describe( '@elementor/user-preferences - useActionProps', () => {
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
```
Comment (Nevoss) [inline packages/packages/user-preferences/src/hooks/__tests__/use-action-props.test.ts:12]: Remove

File: packages/packages/user-preferences/src/hooks/use-action-props.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { __ } from '@wordpress/i18n';
+import { openRoute } from '@elementor/v1-adapters';
+import ToggleRightIcon from '../icons/toggle-right-icon';
+
+// TODO: Use return type from the TopBar package once merged.
```
Comment (Nevoss) [inline packages/packages/user-preferences/src/hooks/use-action-props.ts:5]: ?

File: packages/packages/user-preferences/src/hooks/use-action-props.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+export default function useActionProps() {
+	return {
+		icon: ToggleRightIcon,
+		title: __( 'User Preferences', 'elementor' ),
+		onClick: () => openRoute( 'panel/editor-preferences' ),
```
Comment (Nevoss) [inline packages/packages/user-preferences/src/hooks/use-action-props.ts:10]: Should it be toggle? with state and closed when click again?

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: packages/packages/user-preferences/src/hooks/use-action-props.ts (hunk 3 above)
Comment (StyleShit) [reply packages/packages/user-preferences/src/hooks/use-action-props.ts:12]: Done

File: packages/packages/user-preferences/src/hooks/__tests__/use-action-props.test.ts (hunk 1 above)
Comment (StyleShit) [reply packages/packages/user-preferences/src/hooks/__tests__/use-action-props.test.ts:13]: Done

File: packages/packages/user-preferences/src/hooks/use-action-props.ts (hunk 2 above)
Comment (StyleShit) [reply packages/packages/user-preferences/src/hooks/use-action-props.ts:5]: Done

**Review by Nevoss — DISMISSED** [review]

File: packages/packages/user-preferences/src/icons/toggle-right-icon.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import { SvgIcon, SvgIconProps } from '@elementor/ui';
+
+export default function ToggleRightIcon( props: SvgIconProps ) {
```
Comment (rotemee) [inline packages/packages/user-preferences/src/icons/toggle-right-icon.tsx:4]: Add forwardRef.

File: packages/packages/user-preferences/package.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@elementor/user-preferences",
+  "dependencies": {
+    "@elementor/top-bar": "*",
+    "@elementor/ui": "*",
```
Comment (rotemee) [inline packages/packages/user-preferences/package.json:5]: Remove `@elementor/ui`.

**Review by rotemee — CHANGES_REQUESTED** [review]

File: packages/packages/user-preferences/src/icons/toggle-right-icon.tsx (hunk 4 above)
Comment (StyleShit) [reply packages/packages/user-preferences/src/icons/toggle-right-icon.tsx:4]: Oops, it wasn't even in the right place...
fixed

File: packages/packages/user-preferences/package.json (hunk 5 above)
Comment (StyleShit) [reply packages/packages/user-preferences/package.json:5]: Done

**Review by rotemee — APPROVED** [review]

**Pull request merged**